lib/smart_answer/calculators/statutory_sick_pay_calculator.rb

```diff
@@ -287,6 +311,28 @@ def before_coronavirus_entitlement_date?
         current_piw.begins_on < PeriodOfIncapacityForWork::CORONAVIRUS_SSP_START_DATE
       end
 
+      def ends_after_coronavirus_entitlement_date?
+        current_piw.ends_on > PeriodOfIncapacityForWork::CORONAVIRUS_SSP_START_DATE
+      end
+
+      def before_coronavirus_shield_date?
+        current_piw.begins_on < PeriodOfIncapacityForWork::CORONAVIRUS_SHIELDING_START_DATE
+      end
+
+      def ends_after_coronavirus_shield_date?
+        current_piw.ends_on > PeriodOfIncapacityForWork::CORONAVIRUS_SHIELDING_START_DATE
+      end
+
+      def coronavirus_related_unpaid_workdays_missed(coronavirus_period_start_date, piw_startdate)
+        coronavirus_period_start_date = coronavirus_period_start_date - 1 #We want to count to the date prior
+        dates = piw_startdate..coronavirus_period_start_date
+        matching_dates = []
+        dates.each do |d|
+          matching_dates << d if days_of_the_week_worked.include?(d.wday.to_s)
+        end
+        matching_dates
```

Comment:
```suggestion
        dates.map { |d| d if days_of_the_week_worked.include?(d.wday.to_s) }.compact
```

I think this is a neater way of doing this.

Reply:
Ah good, I wasn't fond of this myself. I'll throw it in locally and test it before committing.